### PR TITLE
Pt 161288395 refactor sc ws apis

### PR DESCRIPTION
**Pull request by velzevur**
[SC WebSocket Versioning](https://www.pivotaltracker.com/story/show/161288395)
SC WS API is splitted into different behaviours, independent of the WebSocket handler

### CODE REVIEW

**Comment by velzevur** [comment]
tests are failing, I am investigating

**Comment by zp-sd** [comment]
Note: I did not do proper code review yet, but I'd expect some commit names to be more descriptive (or commits squashed?), e.g. `WIP` 😉 . Just to remember before final merge.

**Comment by velzevur** [comment]
Yes, commits are going to be squashed and final commit message will be something like `Channels' WebSocket versioning`

File: apps/aehttp/src/sc_ws_api_jsonrpc.erl [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+       , <<"message">> => error_data_msg(C) } || C <- Codes].
+
+%% Mimicking Ethereum suggested custom error codes (not all relevant here)
+error_data_msg(Code) ->
+    maps:get(Code, error_data_msgs(), <<"Unknown error">>).
+
+error_data_msgs() ->
+    #{
+       100 => <<"X doesn't exist">>
+     , 101 => <<"Requires coin">>      %% (Requires ether)
```
Comment (hanssv) [inline apps/aehttp/src/sc_ws_api_jsonrpc.erl]: ether??

**Pull request merged**